scripts/vitest.config.ts

```diff
@@ -0,0 +1,22 @@
+import type { ESBuildOptions } from 'vite';
+import { defineConfig, type UserConfig } from 'vitest/config';
+
+export const createVitestConfig = (options: UserConfig = {}) =>
+	defineConfig({
+		...options,
+		test: {
+			...options?.test,
+			globals: true,
+			coverage: {
+				...options.test?.coverage,
+				enabled: true,
+				reporter: ['text', 'lcov', 'clover'],
+				exclude: [...(options.test?.coverage?.exclude ?? []), '**/node_modules/**', '**/dist/**', '**/tests/**']
```

Comment:
Use [`configDefaults`](https://vitest.dev/config/#configuration:~:text=You%20can%20retrieve%20Vitest%27s%20default%20options%20to%20expand%20them%20if%20needed%3A) instead of `**/node_modules/**` and such

Reply:
I literally copied this from sapphiredev/utilities.